robot_framework/process.py

```diff
@@ -1,8 +1,237 @@
 """This module contains the main process of the robot."""
 
+import os
+import json
+from dataclasses import dataclass
+from io import BytesIO
+import io
+
+from openpyxl import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from openpyxl.styles import Font
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.common.exceptions import NoSuchElementException
 from OpenOrchestrator.orchestrator_connection.connection import OrchestratorConnection
 
+from itk_dev_shared_components.eflyt import eflyt_login, eflyt_search
+from itk_dev_shared_components.graph import mail, authentication
+from itk_dev_shared_components.graph.authentication import GraphAccess
+from itk_dev_shared_components.smtp import smtp_util
+from robot_framework import config
+
+
+@dataclass
+class CprCase:
+    """A dataclass representing a pair of CPR and Case numbers"""
+    case: str
+    cpr: str
+    name: str
+    phone_number: list[str] | None
+
+
+@dataclass
+class EmailInput:
+    '''A dataclass representing input from an email'''
+    cpr_cases: list[CprCase]
+    requester: str
+    email: mail.Email | None
+
 
-def process(orchestrator_connection: OrchestratorConnection) -> None:
+def process(email_data: EmailInput | None, graph_access: GraphAccess, orchestrator_connection: OrchestratorConnection) -> None:
     """Do the primary process of the robot."""
     orchestrator_connection.log_trace("Running process.")
+
+    # Login
+    eflyt_credentials = orchestrator_connection.get_credential(config.EFLYT_LOGIN)
+    browser = eflyt_login.login(eflyt_credentials.username, eflyt_credentials.password)
+
+    if email_data:
+        handle_email(email_data, browser)
+        write_excel(email_data.cpr_cases)
+        send_status_emails(email_data)
+        if email_data.email:
+            mail.delete_email(email_data.email, graph_access)
+        os.remove(config.EMAIL_ATTACHMENT)
+
+
+def handle_email(email_input: EmailInput, browser: webdriver.Chrome) -> None:
+    """Handle an email by looking up each pair of CPR and cases in eflyt and adding a phone number to the instance.
+
+    Args:
+        email_input: An EmailInput object containing a list of CPR/Case pairs.
+        browser: A WebDriver to use selenium.
+    """
+    for cpr_case_pair in email_input.cpr_cases:
+        if cpr_case_pair.phone_number is not None:
+            continue
+        try:
+            eflyt_search.open_case(browser, cpr_case_pair.case)
+        except NoSuchElementException:
```

Comment:
Why do you expect a NoSuchElementException? Maybe a comment if it's something we expect to happen

Reply:
Also why not merge the two try's?